stumpy/motifs.py

```diff
@@ -448,16 +465,21 @@ def match(
     m = Q.shape[1]
     excl_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
 
-    if M_T is None or Σ_T is None:  # pragma: no cover
-        T, M_T, Σ_T = core.preprocess(T, m)
+    if M_T is None or Σ_T is None or T_subseq_isconstant is None:  # pragma: no cover
+        T, M_T, Σ_T, T_subseq_isconstant = core.preprocess(T, m)
```

Comment:
I think `match` is public API, and I should better break down the if-condition into two if-blocks.

Reply:
See comment above